### ISSUE TITLE
rds template refactor: re-use the code in the config provider

### ISSUE THREAD

**Issue by wbpcode** [envoyproxy/envoy#26096]
We extracted the common part of rds to a template in https://github.com/envoyproxy/envoy/pull/18846. By this way, any proxy could implement it's rds by simple code change.

But I found the code of rds template is similar with the code in the `config/config_provider.h`. I am not sure it's possible to re-use it or merge them. Then we can make the xds-related code more clean and unified.

cc @tkovacs-2

**Comment by tkovacs-2** [comment]
Yes, it looks very similar to common/rds. I will check how to reuse.
Just I don't see other usage than in scoped_rds.

**Comment by wbpcode** [comment]
Not sure when will #24773 be land. Seems that should the final pattern.
